hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -875,7 +879,11 @@ public void buildMetadataPartitions(HoodieEngineContext engineContext, List<Hood
     inflightIndexes.addAll(indexPartitionInfos.stream().map(HoodieIndexPartitionInfo::getMetadataPartitionPath).collect(Collectors.toSet()));
     dataMetaClient.getTableConfig().setValue(HoodieTableConfig.TABLE_METADATA_PARTITIONS_INFLIGHT.key(), String.join(",", inflightIndexes));
     HoodieTableConfig.update(dataMetaClient.getFs(), new Path(dataMetaClient.getMetaPath()), dataMetaClient.getTableConfig().getProps());
-    initialCommit(indexUptoInstantTime, partitionTypes);
+
+    String commitInstantTime =
+        metadataMetaClient.getActiveTimeline().containsInstant(indexUptoInstantTime)
+            ? indexUptoInstantTime + METADATA_INDEXER_TIME_SUFFIX : indexUptoInstantTime;
```

Comment:
we should not hit the else block. lets do checkState and fail

Reply:
Fixed.